components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -308,6 +365,92 @@ static esp_err_t set_websocket_transport_optional_settings(esp_websocket_client_
+            if (client->keep_alive_cfg.keep_alive_enable) {
+                esp_transport_tcp_set_keep_alive(tcp, &client->keep_alive_cfg);
+            }
+            if (client->if_name) {
+                esp_transport_tcp_set_interface_name(tcp, client->if_name);
+            }
+
+            esp_transport_handle_t ws = esp_transport_ws_init(tcp);
+            ESP_WS_CLIENT_MEM_CHECK(TAG, ws, {
+                esp_transport_destroy(tcp);
+                return ESP_ERR_NO_MEM;
+            });
+
+            esp_transport_set_default_port(ws, WEBSOCKET_TCP_DEFAULT_PORT);
+            esp_transport_list_add(client->transport_list, ws, WS_OVER_TCP_SCHEME);
+            ESP_WS_CLIENT_ERR_OK_CHECK(TAG, set_websocket_transport_optional_settings(client, WS_OVER_TCP_SCHEME), return ESP_FAIL;)
+        } else if (strcasecmp(client->config->scheme, WS_OVER_TLS_SCHEME) == 0) {
+            esp_transport_handle_t ssl = esp_transport_ssl_init();
+            ESP_WS_CLIENT_MEM_CHECK(TAG, ssl, return ESP_ERR_NO_MEM);
+
+            esp_transport_set_default_port(ssl, WEBSOCKET_SSL_DEFAULT_PORT);
+            esp_transport_list_add(client->transport_list, ssl, "_ssl"); // need to save to transport list, for cleanup
+            if (client->config->use_global_ca_store == true) {
+                esp_transport_ssl_enable_global_ca_store(ssl);
+            } else if (client->config->cert_pem) {
+                if (!client->config->cert_len) {
+                    esp_transport_ssl_set_cert_data(ssl, client->config->cert_pem, strlen(client->config->cert_pem));
+                } else {
+                    esp_transport_ssl_set_cert_data_der(ssl, client->config->cert_pem, client->config->cert_len);
+                }
+            }
+            if (client->config->client_cert) {
+                if (!client->config->client_cert_len) {
+                    esp_transport_ssl_set_client_cert_data(ssl, client->config->client_cert, strlen(client->config->client_cert));
+                } else {
+                    esp_transport_ssl_set_client_cert_data_der(ssl, client->config->client_cert, client->config->client_cert_len);
+                }
+            }
+            if (client->config->client_key) {
+                if (!client->config->client_key_len) {
+                    esp_transport_ssl_set_client_key_data(ssl, client->config->client_key, strlen(client->config->client_key));
+                } else {
+                    esp_transport_ssl_set_client_key_data_der(ssl, client->config->client_key, client->config->client_key_len);
+                }
+            }
+            if (client->config->skip_cert_common_name_check) {
+                esp_transport_ssl_skip_common_name_check(ssl);
+            }
+
+            esp_transport_handle_t wss = esp_transport_ws_init(ssl);
+            ESP_WS_CLIENT_MEM_CHECK(TAG, wss, {
+                esp_transport_destroy(ssl);
```

Comment:
Must call `esp_transport_list_destroy`.

Reply:
The same as https://github.com/espressif/esp-protocols/pull/35#discussion_r874290728